apps/spruce/src/pages/version/Tabs/VersionTiming/index.tsx

```diff
@@ -0,0 +1,128 @@
+  taskCount: number;
+}
+
+const VersionTiming: React.FC<Props> = ({ taskCount, versionId }) => {
+  const dispatchToast = useToastContext();
+  const { search } = useLocation();
+
+  const defaultSort = `${TaskSortCategory.Duration}:${SortDirection.Desc}`;
+  const [queryParams, setQueryParams] = useQueryParams();
+  const versionAnalytics = useVersionAnalytics(versionId);
+  const queryVariables = useQueryVariables(search, versionId);
+  const hasQueryVariables = Object.keys(parseQueryString(search)).length > 0;
+  const hasBuildVariantQueryParam = !!queryParams.variant;
+  const shouldPaginateData = hasBuildVariantQueryParam;
+
+  useEffect(() => {
+    setQueryParams({
+      ...queryParams,
+      [TableQueryParams.Sorts]: defaultSort,
+    });
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const clearQueryParams = () => {
+    setQueryParams({
+      ...queryParams,
+      [PatchTasksQueryParams.TaskName]: undefined,
+      [PatchTasksQueryParams.Variant]: undefined,
+      [PatchTasksQueryParams.Statuses]: undefined,
+      [PatchTasksQueryParams.BaseStatuses]: undefined,
+      [PaginationQueryParams.Page]: undefined,
+      [TableQueryParams.Sorts]: defaultSort,
+    });
+  };
+
+  const { data, loading, refetch, startPolling, stopPolling } = useQuery<
+    VersionTaskDurationsQuery,
+    VersionTaskDurationsQueryVariables
+  >(VERSION_TASK_DURATIONS, {
+    variables: {
+      ...queryVariables,
+      taskFilterOptions: {
+        ...queryVariables.taskFilterOptions,
+        limit:
+          shouldPaginateData && queryVariables.taskFilterOptions.limit
+            ? queryVariables.taskFilterOptions.limit
+            : undefined,
+      },
+    },
+    skip: !hasQueryVariables,
+    pollInterval: DEFAULT_POLL_INTERVAL,
+    onError: (err) => {
+      dispatchToast.error(`Error fetching patch tasks ${err}`);
+    },
+  });
+  usePolling({ startPolling, stopPolling, refetch });
+  const { version } = data || {};
+  const { tasks } = version || {};
+  const { count = 0, data: tasksData = [] } = tasks || {};
+  const { limit, page } = queryVariables.taskFilterOptions;
+
+  const chartData = hasBuildVariantQueryParam
+    ? transformTaskDurationDataToTaskGanttChartData(tasksData)
+    : transformTaskDurationDataToVariantGanttChartData(tasksData);
+
+  return (
+    <>
+      <Body>
+        This page provides a timeline view of{" "}
+        {hasBuildVariantQueryParam ? "task" : "variant"} wall clock duration in
+        the version.
+      </Body>
+      <br />
+      {shouldPaginateData && (
+        <TableControl
+          filteredCount={count}
+          label="tasks"
+          limit={limit || 0}
+          onClear={clearQueryParams}
+          onPageSizeChange={(l) => {
+            versionAnalytics.sendEvent({
+              name: "Changed page size",
+              "page.size": l,
+            });
+          }}
+          page={page || 0}
+          totalCount={taskCount}
+        />
+      )}
```

Comment:
As a general design principle it's kind of jarring when large navigational UI elements are suddenly removed or added to the same view. I think we can keep this consistent and just disable the buttons when they are not modifiable. 


https://github.com/user-attachments/assets/55470c07-d6f4-4c26-b52c-ef5c33ed5cea